basic/atm/atm_exercise.py

```diff
@@ -0,0 +1,181 @@
+    new_balance = curr_balance - amount_to_withdrawal
+    customers_data[customer_id][BALANCE_KEY] = new_balance
+    print(f"withdrawal was successful {customer_id} now has {new_balance}$")
+
+
+def cash_deposit(customers_data, customer_id):
+    """Adds the 'amount_to_deposit' to the customer's balance.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+    """
+    amount_to_deposit = get_a_number("Enter the amount you want to deposit.")
+    new_balance = get_balance(customers_data, customer_id) + amount_to_deposit
+    customers_data[customer_id][BALANCE_KEY] = new_balance
+    print(f"withdrawal was successful {customer_id} now has {new_balance}$")
+
+
+def change_password(customers_data, customer_id):
+    """Changes the given customer's password.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+    """
+    new_pass = input("Enter a new password.\n")
+    customers_data[customer_id][PASS_KEY] = new_pass
+    print(f"Password was changed to {new_pass}")
+
+
+def get_data_from_file(filename):
+    """Gets bank details from file, arranges and returns them in dictionary.
+
+    Args:
+        filename (str): The path to the atm details file.
+
+    Returns (dict): Dictionary of all customer's data from the file.
+    """
+    if os.path.exists(filename):
+        with open(filename, "r") as atm_file:
+            customers = {}
+            for line in atm_file:
+                customer_id, the_pass, balance = line.split()
+                customer = {PASS_KEY: the_pass, BALANCE_KEY: balance}
+                customers[customer_id] = customer
+        return customers
+    else:
+        print(f"the file path : '{filename}' doesn't exist")
+        return -1
+
+
+def switcher(customers_data, customer_id, choice):
+    """Executes function according to the choice given.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+        choice (str): The user's desired function to execute.
+    """
+    if choice == "1":
+        check_balance(customers_data, customer_id)
+    elif choice == "2":
+        cash_withdrawal(customers_data, customer_id)
+    elif choice == "3":
+        cash_deposit(customers_data, customer_id)
+    else:
+        change_password(customers_data, customer_id)
+
+
+def update_file(filename, data):
+    """Updates the file with the data in the given dict 'data'.
+
+    Args:
+        filename (str): Path too the atm file.
+        data (dict): Dictionary with data from the atm file.
+    """
+    with open(filename, "w") as atm_file:
+        for key in data:
+            line = f"{key} {data[key][PASS_KEY]} {data[key][BALANCE_KEY]}\n"
+            atm_file.write(line)
+
+
+def atm(filename):
```

Comment:
this function is too long
a thumb rule - if the function is more then 15~20 lines, its probably do more than 1 logic, and it can be splited to smaller functions